src/errors.ts

```diff
@@ -0,0 +1,12 @@
+export class AuthenticationError extends Error {
+  error: string;
+  error_description: string;
+  state: string;
+
+  constructor(error: string, error_description?: string, state?: string) {
```

Comment:
😬typescript tip of the day 😬 

you can add `pulic` or `private` in the constructor args to automatically create the fields in the class:

```js
class CustomError extends Error {
  constructor(
    public error: string,
    public error_description: string,
    public state: string
  ) {
    super(error_description);
  }
}

```

Reply:
Thanks @luisrudge !